peer/bind.go

```diff
@@ -123,42 +127,46 @@ func (c *BoundChooser) Introspect() introspection.ChooserStatus {
 // BindPeers returns a binder (suitable as an argument to peer.Bind) that
 // binds a peer list to a static list of peers for the duration of its
 // lifecycle.
-func BindPeers(ids []peer.Identifier) Binder {
+func BindPeers(ids []peer.Identifier) peer.Binder {
```

Comment:
The peer.Binder type (which is a function signature), is also the expected type returned by customer BinderSpec Build() functions. Looking in BuildChooser(), you'll see that we use BindPeers() instead of BinderSpec Build() to fill a peer.Binder, when the user provides `peers` instead of `with: custom-binder`.

We should think of a different name than `with`, for example `bind`, or use `peerlist` with URL protocol and get the binder spec name from the protocol scheme, e.g., `peerlist: dns+srv://service.example.com` implies the `dns+srv` binder / peer list updater.

Then we can introduce `"with"` presets to the Configurator separately. That may be a higher level tool.